pkgs/applications/misc/mupdf/default.nix

```diff
@@ -83,15 +84,16 @@ stdenv.mkDerivation rec {
 
     # fix libclang unnamed struct format
     for wrapper in ./scripts/wrap/{cpp,state}.py; do
-      substituteInPlace "$wrapper" --replace 'struct (unnamed' '(unnamed struct'
+      substituteInPlace "$wrapper" \
+        --replace-fail 'struct (unnamed' '(unnamed struct'
     done
   '';
 
   makeFlags = [
     "prefix=$(out)"
     "shared=yes"
+    "USE_SONAME=no"
```

Comment:
You need `env.USE_SONAME = "no"`, the python wrap script doesn't get the memo otherwise and that is (indirectly) causing the pymupdf imports to fail

There's also pymupdf test failures, but I don't have time to dig further right now

Reply:
Do you have an idea why https://github.com/ArtifexSoftware/mupdf/blob/1.24.0/Makefile#L644 doesn't have an effect? The non-version shared library is supposed to be created even without setting `USE_SONAME=no`.

Reply:
The makefile gets the memo, but this snippet does not: https://github.com/ArtifexSoftware/mupdf/blob/21fb0a2bf815c927cf09881f799f78cbece0daf2/scripts/wrap/__main__.py#L1165-L1166

If you don't use the env var, then the libmupdfcpp.so file gets a SONAME that does not exist because the makefile does not link it

Reply:
(it sets soname via linker arg here: <https://github.com/ArtifexSoftware/mupdf/blob/21fb0a2bf815c927cf09881f799f78cbece0daf2/scripts/wrap/__main__.py#L1716>)

Reply:
@lilyinstarlight Let me express myself more clearly. If I don't set `USE_SONAME=no` I expect there to be a versioned shared library and a non-versioned link to it, because of the makefile snippet I shared above. However that link is never created. I don't understand why.

Reply:
> @lilyinstarlight Let me express myself more clearly. If I don't set `USE_SONAME=no` I expect there to be a versioned shared library and a non-versioned link to it, because of the makefile snippet I shared above. However that link is never created. I don't understand why.

Ohhh, yeah that's because it's not added in the `postInstall`, apologies for misunderstanding 😅

https://github.com/NixOS/nixpkgs/blob/4a3db5a7999aff6fb1dfcd9c33a5cfb64dd8770e/pkgs/applications/misc/mupdf/default.nix#L170

Reply:
We don't use the makefile to install the cpp lib because make target dependencies try to rebuild everything as not-static iirc

Reply:
I'm talking about `libmupdf.so`, not `libmupdfcpp.so`.

Reply:
Ah, I really shouldn't be computering when I'm so tired... the makefile doesn't seem to use the `install-shared-c` target when doing `make install` and the `install-libs` target doesn't create the link by itself: https://github.com/ArtifexSoftware/mupdf/blob/1.24.0/Makefile#L478